src/Sema.zig

```diff
@@ -29781,8 +29781,10 @@ fn semaStructFields(mod: *Module, struct_obj: *Module.Struct) CompileError!void
                 extra_index += body.len;
                 const init = try sema.resolveBody(&block_scope, body, struct_obj.zir_index);
                 const field = &struct_obj.fields.values()[i];
-                const coerced = try sema.coerce(&block_scope, field.ty, init, src);
-                const default_val = (try sema.resolveMaybeUndefVal(&block_scope, src, coerced)) orelse
+                const tree = try sema.getAstTree(&block_scope);
+                const init_src = containerFieldInitSrcLoc(decl, tree.*, 0, i);
+                const coerced = try sema.coerce(&block_scope, field.ty, init, init_src);
+                const default_val = (try sema.resolveMaybeUndefVal(&block_scope, init_src, coerced)) orelse
```

Comment:
Finding the accurate source location requires loading and parsing the file and should only be done when the check fails. This is accomplished by first passing `.unneeded` as the source and only getting the accurate source if one of the functions returns `error.NeededSourceLocation`, you can find examples of that being done by grepping for that error in `Sema.zig`

`resolveMaybeUndefVal` only fails if you modify `lib/std/builtin.zig` so I think you can just pass that the less accurate source location. I'd prefer to make it not require a source location at all.

Reply:
> `resolveMaybeUndefVal` only fails if you modify `lib/std/builtin.zig`

I've tried reading the source but I'm having trouble understanding why this is the case. Could you please elaborate what you mean by this?